docs/user_guides/fs/vector_similarity_search.md

```diff
@@ -0,0 +1,96 @@
+---
+description: Users guide about how to use Hopsworks for vector similarity search 
+---
+
+# Introduction
+Vector similarity search is a robust technique enabling the retrieval of similar items based on their embeddings or representations. Its applications range across various domains, from recommendation systems to image similarity and beyond. In Hopsworks, this is facilitated through a vector database, such as Opensearch, which efficiently stores and retrieves relevant embeddings. In this guide, we'll walk you through the process of using Hopsworks for vector similarity search step by step.
+
+# Ingesting Data into the Vector Database
+Hopsworks provides a user-friendly API for writing data to both online and offline feature stores. The example below illustrates the straightforward process of ingesting data into both the vector database and the offline feature store using a single insert method. 
+Currently, Hopsworks supports Opensearch as a vector database. 
+
+First, define the index and embedding features in the vector database. The project index will be used if no index name is provided. 
+
+```aidl
+from hsfs import embedding
+
+# Define the index
+emb = embedding.EmbeddingIndex(index_name=None)
+```
+
+Then, add one or more embedding features to the index. Name and dimension of the embedding features are required for identifying which features should be indexed for k-nearest neighbor (KNN) search. Optionally, you can specify the [similarity function](https://github.com/logicalclocks/feature-store-api/blob/master/python/hsfs/embedding.py#L101).
```

Comment:
We shouldn't link directly to the source code. Instead reference the feature-store-api docs directly. 
Look how we do on the API Reference section on this page: https://github.com/logicalclocks/logicalclocks.github.io/blob/7522cbee5395193f7919053307131a7bb30bb527/docs/user_guides/fs/feature_group/create.md